From 0e0440f6902a5ad08e94d089f6aa757af571f273 Mon Sep 17 00:00:00 2001
From: gbalaji <gbalaji@maxlinear.com>
Date: Tue, 16 Jul 2024 00:14:01 +0800
Subject: [PATCH] WLANRTSYS-83055 fAP's are not up in Repeater/AP2 after client
 mode connection

Issue: Fronthaul vaps are not up in client mode in client mode when auto channel
selection is used.In client mode AP2 must adapt the channel and BW settings
of AP1 and acs must be skipped.But acs is getting triggered during setup interface flow
in client mode due to which the fronthaul vaps are not up.

Fix: Skipped the ACS in setup interface flow.

FIXUP for  WLANRTSYS-81745 :[iwlwav-hostap-ng] Client Mode WAV700 changes porting from legacy hostap to hostap-ng
---
 hostapd/ctrl_iface.c | 2 +-
 src/ap/hostapd.c     | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/hostapd/ctrl_iface.c b/hostapd/ctrl_iface.c
index b001aca06..c1e08bd02 100644
--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -1662,11 +1662,11 @@ static int hostapd_ctrl_iface_update_reload(struct hostapd_data *hapd, char *cmd
 	struct hostapd_config *conf = hapd->iconf;
 	int j;
 
-	hapd->iconf->mxl_conf.is_client_mode = true;
 	hostapd_ctrl_iface_disable(iface);
 
 	hostapd_ctrl_iface_config_update(conf, cmd);
 #ifdef CONFIG_VENDOR_MXL
+	hapd->iconf->mxl_conf.is_client_mode = true;
 	if (os_strstr(cmd, "bandwidth="))
 		mxl_hostapd_backup_user_conf_cw(iface);
 #endif /* CONFIG_VENDOR_MXL */
diff --git a/src/ap/hostapd.c b/src/ap/hostapd.c
index 974e72a32..30189bcea 100644
--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -2653,7 +2653,11 @@ static int setup_interface2(struct hostapd_iface *iface)
 		mxl_hostapd_set_eml_capab(iface);
 #endif /* CONFIG_IEEE80211BE_MXL_MLO */
 
+#ifdef CONFIG_VENDOR_MXL
+		if (iface->conf->acs && !iface->conf->mxl_conf.is_client_mode) {
+#else
 		if (iface->conf->acs) {
+#endif /* CONFIG_VENDOR_MXL */
 			iface->freq = 0;
 			iface->conf->channel = 0;
 		}
-- 
2.43.0

